### PR TITLE
sticky editor implementation - ready for review

### PR DESCRIPTION
**Pull request by burtonator**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/10008

## Description of Changes
- new flag FLAG_STICKY_EDITOR will turn this on.
- Implements the figma design for sticky comments
- Here's a video of what the implementation does (below and in comments)

Some implementation issues:

- I used multiple (small) inline components in CommentStateContext. Otherwise, I'd have to move these files out into the main directory and everything would get smashed together.  If you guys think that's ugly, that's fine but I think we should move this component to a subdirectory. Otherwise, all the sub-components will confuse everything and be just as ugly (if not uglier).

- All of this is behind a feature flag and the normal behavior is tested to work just fine.

https://github.com/user-attachments/assets/47cfe895-bd97-4f77-8a96-4bda80b4eb52

## Test Plan
- You can create a thread, then try to comment on a thread on mobile and desktop.  

## Deployment Plan
- turn on the FLAG_STICKY_EDITOR flag. I tested with the flag on and off and both modes work fine.  

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/Mava/Mava.scss [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #mava {
   .loadButton {
     @media only screen and (max-width: $breakpoint-small-max-px) {
-      bottom: 75px !important;
+      display: none !important;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/Mava/Mava.scss:6]: it should not be visible anymore?

File: packages/commonwealth/client/scripts/views/components/StickEditorContainer/MobileInput.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+    <div className="MobileInput">
+      {avatarURL && (
+        <div className="AvatarBox">
+          <Avatar url={avatarURL} size={32} />
+        </div>
+      )}
+
+      <div className="InputBox">
+        <input
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/StickEditorContainer/MobileInput.tsx:69]: it should be CWTextInput from new_designs

File: packages/commonwealth/client/scripts/views/components/StickEditorContainer/CommentStateContext.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      return <>{activator.activeElement}</>;
+    }
+
+    if (activator.defaultElement) {
+      return <>{activator.defaultElement}</>;
+    }
+
+    return null;
+  },
+);
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/StickEditorContainer/CommentStateContext.tsx:140]: we recently discussed about component file size. I do think we should have separate directory for context and inside we put all utils/hooks/components files related to the context

File: packages/commonwealth/client/scripts/views/components/Mava/Mava.scss (hunk 1 above)
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/components/Mava/Mava.scss:6]: no... we're removing it per @dillchen and I have approval to get it nuked in this PR.  Was slated for the next PR but it conflicts with the sticky editor now.

File: packages/commonwealth/client/scripts/views/components/StickEditorContainer/CommentStateContext.tsx (hunk 3 above)
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/components/StickEditorContainer/CommentStateContext.tsx:140]: Sounds good...

File: packages/commonwealth/client/scripts/views/components/StickEditorContainer/MobileInput.tsx (hunk 2 above)
Comment (burtonator) [reply packages/commonwealth/client/scripts/views/components/StickEditorContainer/MobileInput.tsx:69]: Are you sure? I'm using a raw input on purpose here to control the design.  This thing is a bit sensitive.

File: packages/commonwealth/client/scripts/views/components/Mava/Mava.scss (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/Mava/Mava.scss:6]: so let's remove it completely with script etc as now it looks like you just hiding it

File: packages/commonwealth/client/scripts/views/components/StickEditorContainer/MobileInput.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/StickEditorContainer/MobileInput.tsx:78]: CWTextInput is just a wrapper on raw input with some styling applied that make input feel and look the same across the board

**Review by masvelio — APPROVED** [review]

**Review by Israellund — APPROVED** [review]

**Pull request merged**